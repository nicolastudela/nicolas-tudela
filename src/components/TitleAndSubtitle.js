import React from 'react'
import { Text } from 'uiCommons'
import PropTypes from 'prop-types'

const textColorsByBackgroundColors = color => {
  let customColors = {
    subtitle: 'lightBlue',
    title: 'darkBlue',
  }

  if (color === 'lightBlue') {
    customColors = {
      subtitle: 'black',
      title: 'white',
    }
  } else if (color === 'allWhite') {
    customColors = {
      subtitle: 'white',
      title: 'white',
    }
  }

  return customColors
}

const TitleAndSubtitle = ({ subtitle, title, color, ...rest }) => {
  const colors = textColorsByBackgroundColors(color)
  return (
    <>
      {subtitle && (
        <Text
          variant="h4"
          textAlign="center"
          color={colors.subtitle}
          fontFamily="'EB Garamond', serif"
          fontWeight="500"
          fontStyle="italic"
        >
          {subtitle}
        </Text>
      )}
      <Text
        variant="h1"
        textAlign="center"
        color={colors.title}
        fontWeight="bolder"
        {...rest}
      >
        {title}
      </Text>
    </>
  )
}

TitleAndSubtitle.propTypes = {
  subtitle: PropTypes.string,
  title: PropTypes.string,
  color: PropTypes.string,
}

TitleAndSubtitle.defaultProps = {
  subtitle: null,
  title: '',
  color: null,
}

export default TitleAndSubtitle
