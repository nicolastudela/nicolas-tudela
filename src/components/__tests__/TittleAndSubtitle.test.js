import renderer from 'react-test-renderer'
import TitleAndSubtitle from '../TitleAndSubtitle'

test('TitleAndSubtitlerenders properly', () => {
  const component = renderer.create(TitleAndSubtitle)

  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
