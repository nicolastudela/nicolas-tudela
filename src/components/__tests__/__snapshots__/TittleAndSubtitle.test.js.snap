// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleAndSubtitlerenders properly 1`] = `null`;
